src/structures/APIMessage.js

```diff
@@ -163,7 +190,41 @@ class APIMessage {
       }
     }
 
-    return Promise.all(fileLikes.map(f => this.constructor.resolveFile(f)));
+    this.files = await Promise.all(fileLikes.map(f => this.constructor.resolveFile(f)));
+    return this;
+  }
+
+  /**
+   * Converts this APIMessage into an array of APIMessages for each split content
+   * @returns {APIMessage[]}
+   */
+  split() {
+    if (!this.data) this.resolveData();
+
+    if (!(this.data.content instanceof Array)) return [this];
+
+    const apiMessages = [];
+
+    for (let i = 0; i < this.data.length; i++) {
+      let data;
+      let opt;
+
+      if (i === this.data.content.length - 1) {
+        const changes = { content: this.data.content[i] };
+        data = { ...this.data, ...changes };
+        opt = { ...this.options, ...changes };
+      } else {
+        const changes = { content: this.data.content[i], embed: undefined, embeds: undefined };
+        data = { ...this.data, ...changes };
+        opt = { ...this.options, ...changes, files: undefined };
+      }
```

Comment:
Like I said before, this should match one of `MessageOptions` or `WebhookMessageOptions` just to be correct.

Reply:
This should reflect your earlier request of this?